packages/amplify-category-function/src/provider-utils/awscloudformation/utils/environmentVariables.ts

```diff
@@ -0,0 +1,148 @@
+import path from 'path';
+import _ from 'lodash';
+import { JSONUtilities } from 'amplify-cli-core';
+import { loadFunctionParameters } from './loadFunctionParameters';
+import { categoryName, functionParametersFileName } from '../utils/constants';
+
+const getStoredList = (context: any, resourceName: string) => {
+  const projectBackendDirPath = context.amplify.pathManager.getBackendDirPath();
+  const resourcePath = path.join(projectBackendDirPath, categoryName, resourceName);
+  const functionParameters = loadFunctionParameters(context, resourcePath);
+  return _.get(functionParameters, 'environmentVariableList', []);
+};
+
+const updateStoredList = (context, resourceName, newList) => {
+  const projectBackendDirPath = context.amplify.pathManager.getBackendDirPath();
+  const resourcePath = path.join(projectBackendDirPath, categoryName, resourceName);
+  const functionParameterFilePath = path.join(resourcePath, functionParametersFileName);
+  const functionParameters = JSONUtilities.readJson(functionParameterFilePath) as object;
+  _.set(functionParameters, 'environmentVariableList', newList);
+  JSONUtilities.writeJson(functionParameterFilePath, functionParameters);
+};
+
+const getStoredReferences = (context: any, resourceName: string) => {
+  const projectBackendDirPath = context.amplify.pathManager.getBackendDirPath();
+  const resourcePath = path.join(projectBackendDirPath, categoryName, resourceName);
+  const cfnFileName = `${resourceName}-cloudformation-template.json`;
+  const cfnFilePath = path.join(resourcePath, cfnFileName);
+  const cfnContent = JSONUtilities.readJson(cfnFilePath);
+  return _.get(cfnContent, ['Resources', 'LambdaFunction', 'Properties', 'Environment', 'Variables'], {});
+};
+
+const updateStoredReferences = (context, resourceName, newReferences) => {
```

Comment:
I would combine this function with `updateStoredParameters` to be `updateCloudFormationTemplate` and have it update both the params and the env var refs. These configurations are tightly coupled and one should never be called without the other.

Reply:
Sounds good! Thank you for your comment :)